src/lib/asn1/asn1_time.cpp

```diff
@@ -230,14 +232,14 @@ std::string X509_Time::readable_string() const
    if(time_is_set() == false)
       throw Invalid_State("X509_Time::readable_string: No time set");
 
-   std::string output(24, 0);
-
-   std::sprintf(&output[0], "%04d/%02d/%02d %02d:%02d:%02d UTC",
-                year, month, day, hour, minute, second);
-
-   output.resize(23); // remove trailing null
-
-   return output;
+   // desired format: "%04d/%02d/%02d %02d:%02d:%02d UTC"
+   using namespace std;
+   stringstream out;
```

Comment:
Because I need std:: for setw() and setfill(). I tried to do it without `using` but the lines became too long and the structure was gone. I was even considering a sub-scope for line 238-242 to make `using namespace` even more local.